### PR TITLE
Fix #1811

### LINKED ISSUE #1811: Multiple time give_update_edited_donation hook
## Issue Overview

In Give release 1.8.9, `give_update_edited_donation` hook is executed multiple times.

## Expected Behavior

It should be only executed one time and different ones should be in place for before and after as before 1.8.8.

## Current Behavior

Currently, `give_update_edited_donation` hook multiple time exist so the developer can not able to trigger event for before edit donation and after edited donation.

Before donation update hook:
https://github.com/jaydeeprami/Give/blob/release/1.8.9/includes/admin/payments/actions.php#L82

After donation update hook:
https://github.com/jaydeeprami/Give/blob/release/1.8.9/includes/admin/payments/actions.php#L291

## Possible Solution

As in Give 1.8.9 we are deprecating below.
`give_update_edited_purchase`

Also, We have to depreciate below hook:
`give_updated_edited_purchase` and use new hook `give_updated_edited_donation`

## Todos
- [ ] Tests

### PR DESCRIPTION
**Pull request by jaydeeprami**
## Description

Fix #1811 

## How Has This Been Tested?

I have tested with both hook before edited and after edited donation.

Tested on 4.8 and latest branch release 1.8.9

It will not affects.

## Checklist:
- [X] My code is tested.
- [X] My code follows the WordPress code style.
- [X] My code follows has proper inline documentation.

### CODE REVIEW

File: includes/admin/payments/actions.php [hunk 1]
```diff
@@ -75,7 +75,7 @@ function give_update_payment_details( $data ) {
 	/**
 	 * Fires before updating edited donation.
 	 *
-	 * @since 1.0
+	 * @since 1.8.9
```
Comment (ravinderk) [inline includes/admin/payments/actions.php]: @jaydeeprami Why did you change @since here?

File: includes/admin/payments/actions.php [hunk 2]
```diff
@@ -284,11 +284,11 @@ function give_update_payment_details( $data ) {
 	/**
 	 * Fires after updating edited donation.
 	 *
-	 * @since 1.0
+	 * @since 1.8.9
```
Comment (ravinderk) [inline includes/admin/payments/actions.php]: @jaydeeprami Do not remove @since tag.
You can do like this
@since 1.0
@since 1.8.9 Changes hook name give_update_edited_donation -> give_updated_edited_donation

File: includes/admin/payments/actions.php (hunk 1 above)
Comment (jaydeeprami) [inline includes/admin/payments/actions.php]: @ravinderk , I have done changes as per your suggestion.

File: includes/admin/payments/actions.php (hunk 2 above)
Comment (jaydeeprami) [inline includes/admin/payments/actions.php]: @ravinderk , I have done changes as per your suggestion.
